Unlock `bundler` dependency

anthony-robin opened the pull request:
When migrating to Bundler 2, there is no possibility to install the gem anymore.
We should update the dependency lock to allow Bundler 2.

Details:
- Bump `bundler` dependency to allow version 2

MaximeD commented:
Thank you for this PR!
I see that travis is complaining, we will also need to update the [.travis.yml](https://github.com/MaximeD/gem_updater/blob/910d30b544ad12fd31b7de831355c65ab423db8a/.travis.yml#L3-L4) following [these instructions](https://docs.travis-ci.com/user/languages/ruby/#bundler-20).

Also it seems we need to run `bundle update --bundler` as the `Gemfile.lock` still uses bundler 1 (as explained [here](https://bundler.io/guides/bundler_2_upgrade.html#upgrading-applications-from-bundler-1-to-bundler-2)).

Thanks a lot!

anthony-robin commented:
Thank you for pointing me in the right direction ! CI is now ✅ 

Details:

- Added `gem update --system` in `.travis.yml`
- Updated the `Gemfile.lock` with the corresponding `bundler` version

MaximeD commented:
Awesome, PR merged and new version (4.0.0) has been published to rubygems :raised_hands: 

Thanks again for this PR!
